components/scream/src/share/tests/field_tests.cpp

```diff
@@ -126,6 +126,47 @@ TEST_CASE("field", "") {
     // Trying to reshape into something that the allocation cannot accommodate should throw
     REQUIRE_THROWS (f1.get_reshaped_view<Pack<Real,32>***>());
   }
+
+  // Check whether we can correctly manipulate a field with two scalar types.
+  SECTION ("reshape and reduce with multiple scalar types") {
+    Field<Pack<Real,4>,Device> f1 (fid);
+    f1.allocate_view();
+
+    auto v1d = f1.get_view();
+
+    // Reshape the field to a view of Reals, and fill it with ones.
+    // Here we use the field's layout to determine the extents of the relevant
+    // views, since the view's extents include padding introduced by Packs.
+    auto v3d_1 = f1.get_reshaped_view<Real***>();
+    auto host_v3d_1 = Kokkos::create_mirror_view(v3d_1);
+    auto layout = f1.get_header().get_identifier().get_layout();
+    int num_reals = 0;
+    for (int i = 0; i < layout.dim(0); ++i) {
+      for (int j = 0; j < layout.dim(1); ++j) {
+        for (int k = 0; k < layout.dim(2); ++k, ++num_reals) {
```

Comment:
Notice that, since the scalar type of the field is `Pack<Real,4>` the value of dim(2) will be the number of packs, not the number of reals. This is probably another argument in favor of changing the scalar type inside the Field declaration to be simply `Real`.